### PR TITLE
Added SSL support to Redis configuration.

### PR DESCRIPTION
**Pull request by Andy85de**

### CODE REVIEW

File: src/UserProfileService.Redis/Configuration/RedisConfiguration.cs [hunk 1]
```diff
@@ -50,9 +87,62 @@ public class RedisConfiguration
...
+    /// </summary>
+    public string TieBreaker { get; set; } = "__Booksleeve_TieBreak";
+
+    /// <summary>
+    ///     Whether to attempt to use CLIENT SETINFO to set the library name/version on the connection
+    /// </summary>
+    public bool SetClientLibrary { get; set; } = true;
+
+    /// <summary>
+    ///     Redis protocol to use; see section below
```
Comment (stw914) [inline src/UserProfileService.Redis/Configuration/RedisConfiguration.cs:145]: Which section is refered here?

**Review by stw914 — APPROVED** [review]

File: src/UserProfileService.Redis/Configuration/RedisConfiguration.cs (hunk 1 above)
Comment (Andy85de) [reply src/UserProfileService.Redis/Configuration/RedisConfiguration.cs]: I changed the documentation.

File: src/UserProfileService.Redis/Configuration/RedisConfiguration.cs [hunk 2]
```diff
@@ -50,9 +87,62 @@ public class RedisConfiguration
...
+    /// </summary>
+    public RedisProtocol? Protocol { get; set; }
```
[the remark this answers is not in the document]
Comment (Andy85de) [reply src/UserProfileService.Redis/Configuration/RedisConfiguration.cs:147]: It is the same comment like @stw914 did. I changed it.

File: src/UserProfileService.Redis/Configuration/RedisConfiguration.cs [hunk 3]
```diff
@@ -50,9 +87,62 @@ public class RedisConfiguration
...
+    /// </summary>
+    public int SyncTimeout { get; set; } = 5000;
+
+    /// <summary>
+    /// Time (ms) to allow for asynchronous operations
+    /// </summary>
+    public int AsyncTimeout { get; set; } = 5000;
+
+    /// <summary>
+    /// 	Key to use for selecting a server in an ambiguous primary scenario.
```
[the remark this answers is not in the document]
Comment (Andy85de) [reply src/UserProfileService.Redis/Configuration/RedisConfiguration.cs:135]: I don't see any additional whitespace.

File: src/UserProfileService.Redis/Configuration/RedisConfiguration.cs [hunk 4]
```diff
@@ -50,9 +87,62 @@ public class RedisConfiguration
     /// </summary>
     public string User { get; set; }
     
+    /// <summary>
+    /// Type of proxy in use (if any); for example “twemproxy/envoyproxy”
+    /// </summary>
+    public Proxy Proxy { get; set; } = Proxy.None;
```
[the remark this answers is not in the document]
Comment (Andy85de) [reply src/UserProfileService.Redis/Configuration/RedisConfiguration.cs:93]: I added the possible values to the documentation.

**Review by EmmanuelAv360 — APPROVED** [review]

**Pull request merged**